Mono.Debugging/Mono.Debugging.Evaluation/AsyncOperationManager.cs

```diff
@@ -27,215 +27,172 @@
 
 using System;
 using System.Collections.Generic;
-using ST = System.Threading;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using Mono.Debugging.Client;
 
 namespace Mono.Debugging.Evaluation
 {
-	public class AsyncOperationManager: IDisposable
+	public class AsyncOperationManager : IDisposable
 	{
-		List<AsyncOperation> operationsToCancel = new List<AsyncOperation> ();
-		internal bool Disposing;
-
-		public void Invoke (AsyncOperation methodCall, int timeout)
+		class OperationData
 		{
-			methodCall.Aborted = false;
-			methodCall.Manager = this;
+			public IAsyncOperationBase Operation { get; private set; }
+			public CancellationTokenSource TokenSource { get; private set; }
 
-			lock (operationsToCancel) {
-				operationsToCancel.Add (methodCall);
-				methodCall.Invoke ();
+			public OperationData (IAsyncOperationBase operation, CancellationTokenSource tokenSource)
+			{
+				Operation = operation;
+				TokenSource = tokenSource;
 			}
+		}
 
-			if (timeout > 0) {
-				if (!methodCall.WaitForCompleted (timeout)) {
-					bool wasAborted = methodCall.Aborted;
-					methodCall.InternalAbort ();
-					lock (operationsToCancel) {
-						operationsToCancel.Remove (methodCall);
-						ST.Monitor.PulseAll (operationsToCancel);
-					}
-					if (wasAborted)
```

Comment:
there is no Aborted property on AsyncOperation now, Cancellation is controlled by Task API